core/src/main/java/org/apache/accumulo/core/clientImpl/TabletMergeabilityUtil.java

```diff
@@ -80,23 +80,19 @@ public static String encode(Text split, TabletMergeability tabletMergeability) {
     return gson.toJson(jData);
   }
 
-  public static ByteBuffer encodeAsBuffer(Pair<Text,TabletMergeability> split) {
-    return ByteBuffer.wrap(encode(split).getBytes(UTF_8));
+  public static ByteBuffer encodeAsBuffer(SplitMergeability sm) {
+    return ByteBuffer.wrap(encode(sm.split(), sm.mergeability()).getBytes(UTF_8));
   }
 
-  public static String encode(Pair<Text,TabletMergeability> split) {
-    return encode(split.getFirst(), split.getSecond());
-  }
-
-  public static Pair<Text,TabletMergeability> decode(ByteBuffer data) {
+  public static SplitMergeability decode(ByteBuffer data) {
     return decode(ByteBufferUtil.toString(data));
   }
 
-  public static Pair<Text,TabletMergeability> decode(String data) {
+  public static SplitMergeability decode(String data) {
     GSonData jData = gson.fromJson(data, GSonData.class);
```

Comment:
I think you can use gson with Java records directly. I'm not sure you need to have a separate GSonData class to facilitate the serialization.

Reply:
The GSonData class has different fields than the new record.  There is some translation of what is persisted. However it was easy to convert GSonData to a record, so did that.